lnst/Recipes/ENRT/VirtualOvsBridgeVlanInGuestRecipe.py

```diff
@@ -49,62 +53,58 @@ def test_wide_configuration(self):
         host2.vlan0 = VlanDevice(realdev=host2.eth0, vlan_id=self.params.vlan_id)
         guest1.vlan0 = VlanDevice(realdev=guest1.eth0, vlan_id=self.params.vlan_id)
 
-        configuration = super().test_wide_configuration()
-        configuration.test_wide_devices = [host2.vlan0, guest1.vlan0]
+        config = super().test_wide_configuration()
 
         ipv4_addr = interface_addresses(self.params.net_ipv4, default_start="192.168.10.2/24")
         ipv6_addr = interface_addresses(self.params.net_ipv6, default_start="fc00:0:0:1::2/64")
         for machine in [host2, guest1]:
+            config.configure_and_register_ip(host2.vlan0, next(ipv4_addr))
+            config.configure_and_register_ip(guest1.vlan0, next(ipv6_addr))
             machine.vlan0.ip_add(next(ipv4_addr))
             machine.vlan0.ip_add(next(ipv6_addr))
```

Comment:
This code block does not look right. If this is to replace the `ip_add()` methods, they're still kept here. Also the new `configure_and_register_ip()` is run twice on the same parameters (`host2.vlan0` and `guest1,vlan0`), but should be on `machine.vlan0`.

Reply:
A very good catch. Thanks, fixed.